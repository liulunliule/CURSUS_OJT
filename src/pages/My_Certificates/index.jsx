import React from "react";
import "./index.scss";
import { createGlobalStyle } from "styled-components";
import { UilAward, UilTrashAlt} from "@iconscout/react-unicons";
import { useSelector } from "react-redux";
import { Link } from "react-router-dom";
const GlobalStyles = createGlobalStyle`

  body {
    background-color: #f7f7f7; 
    margin: 0;
    padding: 0;
    color: #333; 
  }
`;
const My_Certificates = () => {
  const isShowAll = useSelector((state) => state.savedCourse.isShowAll);
  return (
    <div className={`wrapper-my-certificates ${isShowAll ? "active" : ""}`}>
      <GlobalStyles />
      <div class="container-fluid">
        <div class="row">
          <div class="col-lg-12">
            <h2 class="my-certificates-title">
              <i class="uil uil-award-my-certificates"></i>{" "}
              <UilAward className="award-icon" size="20" /> My Certificates
            </h2>
          </div>
        </div>
        <div class="row">
          <div class="col-12">
            <div class="mt-10">
              <div class="card-dash-my-cerificate">
                <div class="card-dash-left-my-cerificate">
                  <i class="uil uil-award-new-certificate"></i>
                  <UilAward size="32" color="#686f7a" />
                  <h1>Jump Into New Certificate</h1>
                </div>
                <div class="card-dash-right-my-cerificate">
                  <Link to="/secondLayout/certification_center">
                  <button
                    class="create-btn-dash-my-cerificate"
                    onclick="window.location.href = 'certification_center.html';"
                  >
                    New Certificate
                  </button>
                  </Link>
                </div>
              </div>
              <div class="table-my-cerificate">
                <div class="table-responsive-my-cerificate">
                  <table
                    class="table ucp-table-my-cerificate"
                    id="content-table"
                  >
                    <thead class="thead-s">
                      <tr>
                        <th class="text-left" scope="col">
                          Item No.
                        </th>
                        <th scope="col">Title</th>
                        <th class="text-left" scope="col">
                          Marks
                        </th>
                        <th class="text-left" scope="col">
                          Out Of
                        </th>
                        <th class="text-left" scope="col">
                          Upload Date
                        </th>
                        <th class="text-left" scope="col">
                          Certificate
                        </th>
                        <th class="text-left" scope="col">
                          Controls
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td class="text-center">1</td>
                        <td class="cell-ta">WordPress Certificate</td>
                        <td class="text-center">15</td>
                        <td class="text-center">20</td>
                        <td class="text-center">6 April 2019</td>
                        <td class="text-center text-center-hover">
                          <Link to="/certificate">
                            View
                          </Link>
                        </td>
                        <td class="text-center">
                          <a href="#" title="Delete" class="gray-s">
                            <UilTrashAlt size="18" color="#686f7a" />
                          </a>
                        </td>
                      </tr>
                      <tr>
                        <td class="text-center">2</td>
                        <td class="cell-ta">WordPress Pro Certificate</td>
                        <td class="text-center">14</td>
                        <td class="text-center">20</td>
                        <td class="text-center">4 April 2019</td>
                        <td class="text-center text-center-hover">
                          <Link to="/certificate">
                            View
                          </Link>
                        </td>
                        <td class="text-center">
                          <a href="#" title="Delete" class="gray-s">
                            <UilTrashAlt size="18" color="#686f7a" />
                          </a>
                        </td>
                      </tr>
                      <tr>
                        <td class="text-center">3</td>
                        <td class="cell-ta">HTML CSS Certificate</td>
                        <td class="text-center">18</td>
                        <td class="text-center">20</td>
                        <td class="text-center">3 April 2019</td>
                        <td class="text-center text-center-hover">
                          <Link to="/certificate">
                            View
                          </Link>
                        </td>
                        <td class="text-center">
                          <a href="#" title="Delete" class="gray-s">
                            <UilTrashAlt size="18" color="#686f7a" />
                          </a>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default My_Certificates;
