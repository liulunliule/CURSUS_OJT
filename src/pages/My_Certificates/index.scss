.wrapper-my-certificates {
    padding: 90px 0 0px;
    transition: all 0.3s ease-out;
    margin-left: 250px;
    overflow-x: hidden;
}
.active {
    margin-left: 0;
}

.my-certificates-title {
    font-size: 20px;
    margin-top: 0;
    margin-bottom: 18px;
    font-weight: 500;
    color: #333;
    margin-left: 35px;
}
.uil-award-my-certificates {
    width: 28px;
    height: 54px;
}

.table-my-cerificate {
    margin: 0 30px 30px 0;
}

.create-btn-dash-my-cerificate {
    font-size: 14px;
    font-weight: 500;
    color: #fff;
    background: #ed2a26;
    padding: 0px 20px;
    border-radius: 3px;
    border: 0;
    height: 40px;
}

.create-btn-dash-my-cerificate:hover {
    color: #fff;
    background: #333;
}

.card-dash-my-cerificate {
    background: #fff;
    margin-top: 30px;
    width: 95%;
    border-radius: 3px;
    border: 1px solid #efefef;
    padding: 30px 20px;
    margin-bottom: 50px;
    margin-left: 30px;
    margin-right: 30px;
    float: left;
}
.award-iconaward-icon-new-certificate {
    padding-right: 300px;
    color: #686f7a;
}

.card-dash-left-my-cerificate {
    float: left;
    display: flex;
    padding: 4px 0;
}

.card-dash-right-my-cerificate {
    float: right;
}

.card-dash-left-my-cerificate h1 {
    font-size: 24px;
    color: #333;
    font-weight: 400;
    margin-bottom: 0;
    text-align: left;
    margin-top: 0;
    padding-left: 15px;
}

.card-dash-left-my-cerificate i {
    font-size: 30px;
    margin-right: -1px;
    color: #686f7a;
}

.my-certificates-title .award-icon {
    margin-left: -6px;
    margin-right: 5px;
}

.table-responsive-my-cerificate {
    border-radius: 3px;
    margin-bottom: 12px;
    margin-left: 30px;
    overflow-x: inherit;
}

table.table.ucp-table-my-cerificate {
    margin-bottom: 0px;
}

.ucp-table-my-cerificate {
    height: auto;
    overflow: hidden;
    border-radius: 4px;
}

.ucp-table-my-cerificate thead tfoot {
    font-weight: 500;
    padding: 14px;
    border-radius: 4px;
    color: #333;
}

.ucp-table-my-cerificate thead tr th:first-child {
    border-bottom-left-radius: 4px;
}

.ucp-table-my-cerificate thead tr th {
    color: #333;
    font-size: 14px;
    background-color: #ffecec;
}

.ucp-table-my-cerificate tfoot td {
    color: #fff;
}

.ucp-table.earning__table td {
    padding: 1rem;
}

table {
    margin: 0px auto 0px;
    font-size: 14px;
    width: 100%;
}

.ucp-table-my-cerificate tbody {
    background: #fff;
}

.ucp-table-my-cerificate td,
.ucp-table-my-cerificate th {
    border-top: 1px solid #efefef;
}

.ucp-table-my-cerificate td {
    padding: 1.6rem 0.75rem;
    vertical-align: top;
    border-top: 1px solid #dee2e6;
}

.ucp-table-my-cerificate tbody tr td {
    font-size: 14px;
    vertical-align: middle;
    font-weight: 400;
}
.text-center-hover {
    a {
        display: inline-block;
        padding: 6px 12px;
        transition: color 0.3s ease;
        color: #686f7a;

        &:hover {
            color: #000000;
        }
    }
}
.gray-s:hover {
    filter: brightness(0);
}
.text-left {
    font-weight: 500;
}
