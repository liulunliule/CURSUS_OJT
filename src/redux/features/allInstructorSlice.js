import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

export const fetchAllInstructorData = createAsyncThunk('allInstructor/fetchAllInstructorData', async () => {
  const response = await axios.get('https://66908324c0a7969efd9c4d55.mockapi.io/liveInstructor');
  return response.data;
});

const InstructorSlice = createSlice({
  name: 'allInstructor',
  initialState: {
    data: null,
    status: 'idle',
    error: null,
  },
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchAllInstructorData.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchAllInstructorData.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.data = action.payload;
      })
      .addCase(fetchAllInstructorData.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.error.message;
      });
  },
});

export default InstructorSlice.reducer;
