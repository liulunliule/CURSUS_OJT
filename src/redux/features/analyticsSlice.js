// src/features/analyticsSlice.js
import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

export const fetchAnalyticsData = createAsyncThunk('analytics/fetchAnalyticsData', async () => {
  const response = await axios.get('https://66952ff94bd61d8314ca672b.mockapi.io/year/1');
  return response.data;
});

const analyticsSlice = createSlice({
  name: 'analytics',
  initialState: {
    data: null,
    status: 'idle',
    error: null,
  },
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchAnalyticsData.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchAnalyticsData.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.data = action.payload;
      })
      .addCase(fetchAnalyticsData.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.error.message;
      });
  },
});

export default analyticsSlice.reducer;
