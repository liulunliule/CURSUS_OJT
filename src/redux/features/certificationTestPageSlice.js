import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

// Async thunk for fetching questions
export const fetchQuestions = createAsyncThunk(
  'certificationTestPage/fetchQuestions',
  async () => {
    const response = await fetch('https://66753c2ba8d2b4d072ef3683.mockapi.io/Test');
    if (!response.ok) {
      throw new Error('Network response was not ok');
    }
    return await response.json();
  }
);

const certificationTestPageSlice = createSlice({
  name: 'certificationTestPage',
  initialState: {
    questions: [],
    status: 'idle',
    error: null,
  },
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchQuestions.pending, (state) => {
        state.status = 'loading'; // Set loading state
      })
      .addCase(fetchQuestions.fulfilled, (state, action) => {
        state.status = 'succeeded'; // Set succeeded state
        state.questions = action.payload; // Set questions data
      })
      .addCase(fetchQuestions.rejected, (state, action) => {
        state.status = 'failed'; // Set failed state
        state.error = action.error.message; // Set error message
      });
  },
});

export default certificationTestPageSlice.reducer;
