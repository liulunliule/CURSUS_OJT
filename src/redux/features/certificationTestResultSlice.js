import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

// Async thunk for fetching results
export const fetchResult = createAsyncThunk(
  'certificationTestResult/fetchResult',
  async () => {
    try {
      const response = await axios.get('https://66753c2ba8d2b4d072ef3683.mockapi.io/Result/1');
      return response.data;
    } catch (error) {
      throw new Error('Network response was not ok');
    }
  }
);

const certificationTestResultSlice = createSlice({
  name: 'certificationTestResult',
  initialState: {
    result: {},
    status: 'idle',
    error: null,
  },
  reducers: {},
  extraReducers: (builder) => {
    builder
      .addCase(fetchResult.pending, (state) => {
        state.status = 'loading';
      })
      .addCase(fetchResult.fulfilled, (state, action) => {
        state.status = 'succeeded';
        state.result = action.payload;
      })
      .addCase(fetchResult.rejected, (state, action) => {
        state.status = 'failed';
        state.error = action.error.message;
      });
  },
});

export default certificationTestResultSlice.reducer;
