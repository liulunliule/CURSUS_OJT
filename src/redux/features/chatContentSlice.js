import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

export const fetchChatContent = createAsyncThunk(
  'chatContent/fetchChatContent',
  async () => {
    const response = await fetch('https://66908324c0a7969efd9c4d55.mockapi.io/chatContent');
    return response.json();
  }
);

const chatContentSlice = createSlice({
  name: 'chatContent',
  initialState: {
    chatContent: [],
    loading: false,
    error: null,
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchChatContent.pending, (state) => {
        state.loading = true;
        state.error = null;
      })
      .addCase(fetchChatContent.fulfilled, (state, action) => {
        state.loading = false;
        state.chatContent = action.payload;
      })
      .addCase(fetchChatContent.rejected, (state, action) => {
        state.loading = false;
        state.error = action.error.message;
      });
  },
});

export default chatContentSlice.reducer;