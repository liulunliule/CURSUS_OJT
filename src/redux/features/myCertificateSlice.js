import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
import axios from "axios";

export const fetchCertificates = createAsyncThunk(
  "certificate/fetchCertificates",
  async (userId) => {
    const response = await axios.get(
      `https://6696231a0312447373c1386e.mockapi.io/user/${userId}/certificate`
    );
    return response.data;
  }
);

// Uncomment the following code if you want to enable the delete certificate functionality

// export const fetchDeleteMyCertificate = createAsyncThunk(
//   "certificate/fetchDeleteMyCertificate",
//   async (id, { dispatch }) => {
//     await axios.delete(
//       `https://6690897fc0a7969efd9c6a33.mockapi.io/MyCertificate/${id}`
//     );
//     dispatch(deleteCertificate(id));
//   }
// );

export const fetchUpdateMyCertificate = createAsyncThunk(
  "myProfile/fetchUpdateMyCertificate",
  async ({ id, UpdateMyCertificate, userId }) => {
    const response = await axios.put(
      `https://6696231a0312447373c1386e.mockapi.io/user/${userId}/certificate/${id}`,
      UpdateMyCertificate
    );
    return response.data;
  }
);

const certificateSlice = createSlice({
  name: "certificate",
  initialState: {
    mycertificate: [],
    status: "idle",
    error: null,
  },
  reducers: {
    // Uncomment the following code if you want to enable the delete certificate functionality

    // deleteCertificate: (state, action) => {
    //   state.mycertificate = state.mycertificate.filter(
    //     (certificate) => certificate.id !== action.payload
    //   );
    // },
    UpdateMyCertificate: (state, action) => {
      const index = state.mycertificate.findIndex(
        (certificate) => certificate.id === action.payload.id
      );
      if (index !== -1) {
        state.mycertificate[index] = action.payload;
      }
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(fetchCertificates.pending, (state) => {
        state.status = "loading";
      })
      .addCase(fetchCertificates.fulfilled, (state, action) => {
        state.status = "succeeded";
        state.mycertificate = action.payload;
      })
      .addCase(fetchCertificates.rejected, (state, action) => {
        state.status = "failed";
        state.error = action.error.message;
      })
      .addCase(fetchUpdateMyCertificate.fulfilled, (state, action) => {
        const index = state.mycertificate.findIndex(
          (certificate) => certificate.id === action.payload.id
        );
        if (index !== -1) {
          state.mycertificate[index] = action.payload;
        }
      });
  },
});

export const { UpdateMyCertificate } = certificateSlice.actions;

export default certificateSlice.reducer;
