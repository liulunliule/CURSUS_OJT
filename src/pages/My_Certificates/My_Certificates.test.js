import React from "react";
// Các hàm hỗ trợ từ thư viện Testing Library
import { render, screen, waitFor } from "@testing-library/react";
// Thư viện mô phỏng các hành vi của người dùng
import userEvent from "@testing-library/user-event";
// Cung cấp Redux store cho các component
import { Provider } from "react-redux";
// Tạo store Redux với cấu hình mặc định
import { configureStore } from "@reduxjs/toolkit";
// Cung cấp bối cảnh định tuyến cho các component
import { MemoryRouter as Router } from "react-router-dom";
// Component cần kiểm tra
import My_Certificates from "../My_Certificates";
// Các action để kiểm tra
import { fetchCertificates, fetchUpdateMyCertificate } from "../../redux/features/myCertificateSlice";
// Thư viện thông báo toast
import { toast } from "react-toastify";
// Thư viện mở rộng các matcher cho Jest
import "@testing-library/jest-dom";

// Mô phỏng các thông báo toast
jest.mock("react-toastify", () => ({
  toast: {
    success: jest.fn(),
    error: jest.fn(),
  },
}));

// Mô phỏng các action từ Redux
jest.mock("../../redux/features/myCertificateSlice", () => ({
  fetchCertificates: jest.fn(),
  fetchUpdateMyCertificate: jest.fn(),
}));

// Tạo một store giả định với dữ liệu mẫu
const mockStore = {
  certificate: {
    mycertificate: [
      {
        id: "1",
        title: "Certificate 1",
        marks: "90",
        outof: "100",
        date: "2024-01-01",
        statusCertificate: true,
      },
    ],
    status: "succeeded",
  },
  savedCourse: {
    isShowAll: false,
  },
};

// Hàm tạo store Redux cho các bài kiểm tra
const createTestStore = (initialState) => configureStore({
  reducer: (state = initialState) => state,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware(),
});

describe("Component My_Certificates", () => {
  beforeEach(() => {
    // Mô phỏng hành vi của fetchCertificates
    fetchCertificates.mockImplementation(() => async (dispatch) => {
      dispatch({ type: 'fetchCertificates/fulfilled' });
      return Promise.resolve();
    });

    // Mô phỏng hành vi của fetchUpdateMyCertificate
    fetchUpdateMyCertificate.mockImplementation(() => async (dispatch) => {
      dispatch({ type: 'fetchUpdateMyCertificate/fulfilled' });
      toast.success("Delete successfully!!!");
      return Promise.resolve();
    });
  });

  test("displays certificate data when loading is complete", async () => {
    const store = createTestStore(mockStore);

    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <My_Certificates />
        </Router>
      </Provider>
    );

    // Kiểm tra xem dữ liệu chứng chỉ có hiển thị khi tải xong không
    await waitFor(() => {
      expect(screen.getByText("Certificate 1")).toBeInTheDocument();
    });

    // Kiểm tra xem snapshot của component có khớp không
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls fetchCertificates on component mount", async () => {
    const store = createTestStore(mockStore);

    render(
      <Provider store={store}>
        <Router>
          <My_Certificates />
        </Router>
      </Provider>
    );

    // Kiểm tra xem fetchCertificates có được gọi khi component mount không
    await waitFor(() => {
      expect(fetchCertificates).toHaveBeenCalled();
    });
  });

  test("calls fetchUpdateMyCertificate when delete button is clicked", async () => {
    const store = createTestStore(mockStore);

    render(
      <Provider store={store}>
        <Router>
          <My_Certificates />
        </Router>
      </Provider>
    );

    // Kiểm tra xem nút xóa có hiển thị không
    await waitFor(() => {
      const deleteButton = screen.getByTitle("Delete");
      expect(deleteButton).toBeInTheDocument();
    });

    // Mô phỏng hành vi nhấp chuột vào nút xóa
    userEvent.click(screen.getByTitle("Delete"));

    // Kiểm tra xem fetchUpdateMyCertificate có được gọi không
    await waitFor(() => {
      expect(fetchUpdateMyCertificate).toHaveBeenCalled();
    });
  });

  // test("displays loading spinner while loading", () => {
  //   const loadingStore = createTestStore({
  //     ...mockStore,
  //     certificate: { ...mockStore.certificate, status: "loading" },
  //   });

  //   render(
  //     <Provider store={loadingStore}>
  //       <Router>
  //         <My_Certificates />
  //       </Router>
  //     </Provider>
  //   );

  //   // Kiểm tra xem spinner tải có hiển thị khi đang tải không
  //   expect(screen.getByRole("status")).toBeInTheDocument();
  // });
});
