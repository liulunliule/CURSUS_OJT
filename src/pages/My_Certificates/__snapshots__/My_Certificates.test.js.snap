// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component My_Certificates displays certificate data when loading is complete 1`] = `
<DocumentFragment>
  <div
    class="wrapper-my-certificates "
    style=""
  >
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12"
        >
          <h2
            class="my-certificates-title"
          >
            <i
              class="uil uil-award-my-certificates"
            />
             
            <svg
              class="award-icon"
              fill="currentColor"
              height="20"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.87,17.25l-2.71-4.68A6.9,6.9,0,0,0,19,9.25a7,7,0,0,0-14,0,6.9,6.9,0,0,0,.84,3.32L3.13,17.25A1,1,0,0,0,4,18.75l2.87,0,1.46,2.46a1,1,0,0,0,.18.22,1,1,0,0,0,.69.28h.14a1,1,0,0,0,.73-.49L12,17.9l1.93,3.35a1,1,0,0,0,.73.48h.14a1,1,0,0,0,.7-.28.87.87,0,0,0,.17-.21l1.46-2.46,2.87,0a1,1,0,0,0,.87-.5A1,1,0,0,0,20.87,17.25ZM9.19,18.78,8.3,17.29a1,1,0,0,0-.85-.49l-1.73,0,1.43-2.48a7,7,0,0,0,3.57,1.84ZM12,14.25a5,5,0,1,1,5-5A5,5,0,0,1,12,14.25Zm4.55,2.55a1,1,0,0,0-.85.49l-.89,1.49-1.52-2.65a7.06,7.06,0,0,0,3.56-1.84l1.43,2.48Z"
              />
            </svg>
             My Certificates
          </h2>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <div
            class="mt-10"
          >
            <div
              class="card-dash-my-cerificate"
            >
              <div
                class="card-dash-left-my-cerificate"
              >
                <i
                  class="uil uil-award-new-certificate"
                />
                <svg
                  fill="#686f7a"
                  height="32"
                  viewBox="0 0 24 24"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.87,17.25l-2.71-4.68A6.9,6.9,0,0,0,19,9.25a7,7,0,0,0-14,0,6.9,6.9,0,0,0,.84,3.32L3.13,17.25A1,1,0,0,0,4,18.75l2.87,0,1.46,2.46a1,1,0,0,0,.18.22,1,1,0,0,0,.69.28h.14a1,1,0,0,0,.73-.49L12,17.9l1.93,3.35a1,1,0,0,0,.73.48h.14a1,1,0,0,0,.7-.28.87.87,0,0,0,.17-.21l1.46-2.46,2.87,0a1,1,0,0,0,.87-.5A1,1,0,0,0,20.87,17.25ZM9.19,18.78,8.3,17.29a1,1,0,0,0-.85-.49l-1.73,0,1.43-2.48a7,7,0,0,0,3.57,1.84ZM12,14.25a5,5,0,1,1,5-5A5,5,0,0,1,12,14.25Zm4.55,2.55a1,1,0,0,0-.85.49l-.89,1.49-1.52-2.65a7.06,7.06,0,0,0,3.56-1.84l1.43,2.48Z"
                  />
                </svg>
                <h1>
                  Jump Into New Certificate
                </h1>
              </div>
              <div
                class="card-dash-right-my-cerificate"
              >
                <a
                  href="/secondLayout/certification_center"
                >
                  <button
                    class="create-btn-dash-my-cerificate"
                  >
                    New Certificate
                  </button>
                </a>
              </div>
            </div>
            <div
              class="table-my-cerificate"
            >
              <div
                class="table-responsive-my-cerificate"
              >
                <table
                  class="table ucp-table-my-cerificate"
                  id="content-table"
                >
                  <thead
                    class="thead-s"
                  >
                    <tr>
                      <th
                        class="text-left"
                        scope="col"
                      >
                        Item No.
                      </th>
                      <th
                        scope="col"
                      >
                        Title
                      </th>
                      <th
                        class="text-left"
                        scope="col"
                      >
                        Marks
                      </th>
                      <th
                        class="text-left"
                        scope="col"
                      >
                        Out Of
                      </th>
                      <th
                        class="text-left"
                        scope="col"
                      >
                        Upload Date
                      </th>
                      <th
                        class="text-left"
                        scope="col"
                      >
                        Certificate
                      </th>
                      <th
                        class="text-left"
                        scope="col"
                      >
                        Controls
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        class="text-center"
                      >
                        1
                      </td>
                      <td
                        class="cell-ta"
                      >
                        Certificate 1
                      </td>
                      <td
                        class="text-center"
                      >
                        90
                      </td>
                      <td
                        class="text-center"
                      >
                        100
                      </td>
                      <td
                        class="text-center"
                      >
                        2024-01-01
                      </td>
                      <td
                        class="text-center text-center-hover"
                      >
                        <a
                          href="/certificate"
                        >
                          View
                        </a>
                      </td>
                      <td
                        class="text-center"
                      >
                        <a
                          class="gray-s"
                          href="#"
                          title="Delete"
                        >
                          <svg
                            fill="#686f7a"
                            height="18"
                            viewBox="0 0 24 24"
                            width="18"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M10,18a1,1,0,0,0,1-1V11a1,1,0,0,0-2,0v6A1,1,0,0,0,10,18ZM20,6H16V5a3,3,0,0,0-3-3H11A3,3,0,0,0,8,5V6H4A1,1,0,0,0,4,8H5V19a3,3,0,0,0,3,3h8a3,3,0,0,0,3-3V8h1a1,1,0,0,0,0-2ZM10,5a1,1,0,0,1,1-1h2a1,1,0,0,1,1,1V6H10Zm7,14a1,1,0,0,1-1,1H8a1,1,0,0,1-1-1V8H17Zm-3-1a1,1,0,0,0,1-1V11a1,1,0,0,0-2,0v6A1,1,0,0,0,14,18Z"
                            />
                          </svg>
                        </a>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
